### PR TITLE
Update go-ipa & other nits

### PR DESCRIPTION
**Pull request by jsign**
This PR updates `go-ipa` to include two changes that I've done there:
- https://github.com/crate-crypto/go-ipa/pull/53
- https://github.com/crate-crypto/go-ipa/pull/54

The goal of those PRs was to:
- Remove a lot of surface area to maintain and audit in the future.
- Being able to quickly leverage any further improvements the gnark team can make to primitives.

I want to merge this before opening another PR in `go-ipa` with API improvements.

Also, I made some improvements to our CI and README.

TODO:
- [x] Double-check correctness (i.e: state roots unchanged) with an overlay tree bench run.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,23 +1,33 @@
...
 [![DeepSource](https://deepsource.io/gh/gballet/go-verkle.svg/?label=active+issues&show_trend=true&token=OjuF5Q2HbKzpWY8LgWuffNZp)](https://deepsource.io/gh/gballet/go-verkle/?ref=repository-badge)
 [![goreports](https://goreportcard.com/badge/github.com/gballet/go-verkle)](https://goreportcard.com/report/github.com/gballet/go-verkle)
 [![API Reference](https://camo.githubusercontent.com/915b7be44ada53c290eb157634330494ebe3e30a/68747470733a2f2f676f646f632e6f72672f6769746875622e636f6d2f676f6c616e672f6764646f3f7374617475732e737667)](https://pkg.go.dev/github.com/gballet/go-verkle)
 [![Block replay](https://github.com/gballet/go-verkle/actions/workflows/block_replay.yml/badge.svg)](https://github.com/gballet/go-verkle/actions/workflows/block_replay.yml)
 
+
 # go-verkle
 
-An implementation of [Verkle trees](https://dankradfeist.de/ethereum/2021/06/18/verkle-trie-for-eth1.html). When production-ready, the code is to be merged back into go-ethereum.
+> A Go implementation of Verkle Tree datastructure defined in the [spec](https://github.com/crate-crypto/verkle-trie-ref/tree/master/verkle).
```
Comment (jsign) [inline README.md:11]: I think this is a more future-proof reference?

File: README.md [hunk 2]
```diff
@@ -1,23 +1,33 @@
...
+go test ./... -bench=. -run=none -benchmem
 ```
 
-### Benchmarks
+## Security
 
-```
-$ go test . -bench Bench
-```
+If you find any security vulnerability, please don't open a GH issue and contact repo owners directly.
```
Comment (jsign) [inline README.md:28]: I don't know if we want a reference email which would be way better than making the person figure out who (and how) to contact repo owners.

File: README.md [hunk 3]
```diff
@@ -1,23 +1,33 @@
...
 [![API Reference](https://camo.githubusercontent.com/915b7be44ada53c290eb157634330494ebe3e30a/68747470733a2f2f676f646f632e6f72672f6769746875622e636f6d2f676f6c616e672f6764646f3f7374617475732e737667)](https://pkg.go.dev/github.com/gballet/go-verkle)
 [![Block replay](https://github.com/gballet/go-verkle/actions/workflows/block_replay.yml/badge.svg)](https://github.com/gballet/go-verkle/actions/workflows/block_replay.yml)
 
+
 # go-verkle
 
-An implementation of [Verkle trees](https://dankradfeist.de/ethereum/2021/06/18/verkle-trie-for-eth1.html). When production-ready, the code is to be merged back into go-ethereum.
+> A Go implementation of Verkle Tree datastructure defined in the [spec](https://github.com/crate-crypto/verkle-trie-ref/tree/master/verkle). 
 
-Node width is set to 256 children.
```
Comment (jsign) [inline README.md:11]: Removed this since it's contained when we say "we implement the spec".

File: README.md
Comment (jsign) [inline README.md:1]: I refreshed the README a bit.

File: README.md (hunk 1 above)
Comment (gballet) [reply README.md:11]: is it? My money is on Dankrad outliving github :wink:

**Review by gballet — APPROVED** [review]
LGTM

**Pull request merged**